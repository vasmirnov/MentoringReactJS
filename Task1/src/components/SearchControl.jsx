import React from 'react';
import * as s from './searchcontrol.css';
import { OptionText } from './OptionText';

export class SearchControl extends React.PureComponent {
	render() {
		return (
			<div className={s.main}>
				<div className={s.hTitle}>netflixroulette</div>
				<div className={s.h1}>Find your movie</div>
				<input className={s.searchText}></input>
				<div className={s.h2}>
					<div className={s.searchBy}>Search by 
						<OptionText selected={this.props.searchBy} value='title' handler={this.props.searchByHandler} baseClass={s.option} selectedClass={s.selected}/>
						<OptionText selected={this.props.searchBy} value='director' handler={this.props.searchByHandler}  baseClass={s.option} selectedClass={s.selected}/>
					</div>
					<button className={s.searchButton}>Search</button>
				</div>
			</div>
		)
	}
};