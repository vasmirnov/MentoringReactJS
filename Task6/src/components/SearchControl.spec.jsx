import React from 'react';
import renderer from 'react-test-renderer';
import { SearchControl, SEARCH_BY } from './SearchControl';
import { Map } from 'immutable';

jest.disableAutomock();
jest.mock('./OptionText', () => { return { OptionText: 'OptionText' } });
jest.mock('./searchcontrol.css', () => {
    return {
        main: 'main',
        hTitle: 'hTitle',
        h1: 'h1',
        h2: 'h2',
        searchText: 'searchText',
        searchBy: 'searchBy',
        searchButton: 'searchButton',
        option: 'option',
        selected: 'selected'
    }
});

let props;

describe('SearchControl', () => {
    beforeEach(() => {
        props = {
            searchText: 'Test text',
            searchBy: SEARCH_BY,
            searchHandler: () => { },
            searchByHandler: () => { }
        }
    });

    test('render correctly', () => {
        const instance = renderer.create(
            <SearchControl {...props} />
        );
        expect(instance.toJSON()).toMatchSnapshot();
    });
});