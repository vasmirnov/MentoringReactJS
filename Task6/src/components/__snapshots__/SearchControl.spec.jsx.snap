// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchControl render correctly 1`] = `
<div
  className="main"
>
  <div
    className="hTitle"
  >
    netflixroulette
  </div>
  <div
    className="h1"
  >
    Find your movie
  </div>
  <input
    className="searchText"
    onChange={[Function]}
    value="Test text"
  />
  <div
    className="h2"
  >
    <div
      className="searchBy"
    >
      Search by
      <OptionText
        baseClass="option"
        default={true}
        handler={[Function]}
        selected="title"
        selectedClass="selected"
        value="title"
      />
      <OptionText
        baseClass="option"
        handler={[Function]}
        selected="title"
        selectedClass="selected"
        value="director"
      />
    </div>
    <button
      className="searchButton"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
